dds_web/security/auth.py

```diff
@@ -69,21 +69,50 @@ def get_user_roles_common(user):
 
 
 def verify_token_no_data(token):
-    user, claims = __verify_general_token(token)
+    claims = __verify_general_token(token)
+    user = user_from_subject(claims.get("sub"))
     del claims
     gc.collect()
     return user
 
 
+def verify_invite_token_no_sensdata(token):
+    email, TKEK = verify_invite_token(token)
+    del TKEK
+    gc.collect()
+    return email
+
+
+def verify_invite_token(token):
```

Comment:
```
def __base_verify_token_for_invite(token):
    claims = __verify_general_token(token)
    if claims.get("sub"):
        raise AuthenticationError(message="Invalid token")
    return claims


def verify_invite_token(token):
    claims = __base_verify_token_for_invite(token)
    email = claims.get("inv")
    if email:
        return email, models.Invite.query.filter(models.Invite.email == email).first()
    return None, None


def matching_email_with_invite(token, email):
    claims = __base_verify_token_for_invite(token)
    return claims.get("inv") == email
```

Reply:
Replace the function with these three.

Reply:
An invite token should not have a sub, because it doesn't belong to a user yet. Sensitive content is meant to be for one variable which could be a password, key or other sensitive content. There is additional_claims option for a token. We can come up with our own claims here. If in the future there seems to be a reason to put more than one sensitive content in a token, it can go into additional_claims. I don't consider an email sensitive content, but nevertheless every part of our encrypted token is encrypted so if we put an email in there it will be protected. inv does not seem to be a reserved claim name. We could add the email as inv. So an invite token would look something like (I skipped the exp for the sake of example):
```
encrypted_jwt_token(
                username="",
                sensitive_content="BOGUS",
                additional_claims={"inv": "existing_invite_email@mailtrap.io"},
            )
```